api/src/main/java/org/infinispan/api/reactive/KeyValueStore.java

```diff
@@ -0,0 +1,50 @@
+package org.infinispan.api.reactive;
+
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+
+import org.reactivestreams.Publisher;
+
+/**
+ * A Reactive Key Value Store provides a highly concurrent and distributed data structure, non blocking and using
+ * reactive streams.
+ * <p>
+ * TODO: Add more javadoc and examples
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ * @see <a href="http://www.infinispan.org">Infinispan documentation</a>
+ * @since 10.0
+ */
+public interface KeyValueStore<K, V> {
+
+   CompletionStage<V> get(K key);
+
+   CompletionStage<Void> insert(K key, V value);
+
+   CompletionStage<Void> save(K id, V value);
```

Comment:
What's the difference between insert and save? Both have same arguments and have exactly the same return types :\

Reply:
insert is like a putIfAbsent and save is a put

Reply:
If insert is a putIfAbsent, I wonder if it might have some value to return a boolean indicating whether the if the put did put something or not...